examples/mining_cfd.py

```diff
@@ -0,0 +1,125 @@
+"""CFD mining example with visualization using Desbordante algorithms."""
+
+from collections import defaultdict
+from functools import reduce
+import operator
+
+import desbordante
+import pandas
+import numpy as np
+
+# CFD Discovery parameters:
+TABLEPATH = 'examples/datasets/play_tennis.csv'
+MINIMUM_SUPPORT = 8
+MINIMUM_CONFIDENCE = 0.7
+MAXIMUM_LHS_COUNT = 3
+
+
+GREEN_BG_CODE = '\033[1;42m'
+RED_BG_CODE = '\033[1;41m'
+DEFAULT_BG_CODE = '\033[1;49m'
+
+
+def items_to_indices(items):
+    return [item.attribute for item in items]
+
+
+def find_var_attributes(items):
+    result = []
+    for item in items:
+        if item.value is None:
+            result.append(item.attribute)
+    return result
+
+
+def get_supported_mask(pattern, table):
+    table_height = table.shape[0]
+    def get_mask(cfd_item):
+        if cfd_item.value is None:
+            true_array = np.full(
+                shape=table_height,
+                fill_value=True,
+                dtype=np.bool_
+            )
+            return pandas.Series(true_array)
+        columnName = table.columns[cfd_item.attribute]
+        return table[columnName].astype(str).str.lower() == cfd_item.value.lower()
+
+    return reduce(operator.and_, map(get_mask, pattern))
```

Comment:
```suggestion
    table_height = table.shape[0]
    true_array = np.full(shape=table_height, fill_value=True, dtype=np.bool_)
    mask = pandas.Series(true_array)
    for cfd_item in pattern:
        item_value = cfd_item.value
        if item_value is None:
            continue
        mask &= table.columns[cfd_item.attribute].astype(str).str.lower() == item_value.lower()
    return mask
```
But if you really want to write it in a functional style, use [`filter`](https://docs.python.org/3/library/functions.html#filter) and an initializer for `reduce` for completeness.

Reply:
`mask &= table.columns[cfd_item.attribute].astype(str).str.lower() == item_value.lower()` doesn't work, I left it as it was for now, with "column_name" in my latest commit

Reply:
right, I meant `table[table.columns[cfd_item.attribute]]`, but splitting is better.